src/worker/git.rs

```diff
@@ -74,6 +75,98 @@ pub fn update_crate_index(crate_name: String) -> Job {
     Job::IndexSyncToHttp(IndexSyncToHttpJob { crate_name })
 }
 
+/// Regenerates or removes an index file for a single crate
+#[instrument(skip_all, fields(krate.name = ?krate))]
+pub fn sync_to_git_index(
+    env: &Environment,
+    conn: &mut PgConnection,
+    krate: &str,
+) -> Result<(), PerformError> {
+    info!("Syncing to git index");
+
+    let new = get_index_data(krate, conn).context("Failed to get index data")?;
+
+    let repo = env.lock_index()?;
+    let dst = repo.index_file(krate);
+
+    // Read the previous crate contents
+    let old = match fs::read_to_string(&dst) {
+        Ok(content) => Some(content),
+        Err(error) if error.kind() == ErrorKind::NotFound => None,
+        Err(error) => return Err(error.into()),
+    };
+
+    match (old, new) {
+        (None, Some(new)) => {
+            fs::create_dir_all(dst.parent().unwrap())?;
+            let mut file = File::create(&dst)?;
+            file.write_all(new.as_bytes())?;
+            repo.commit_and_push(&format!("Creating crate `{}`", krate), &dst)?;
+        }
+        (Some(old), Some(new)) if old != new => {
+            let mut file = File::create(&dst)?;
+            file.write_all(new.as_bytes())?;
+            repo.commit_and_push(&format!("Updating crate `{}`", krate), &dst)?;
```

Comment:
> This will produce less specific git commit messages than we currently use.

Yep, but I figured since we can't be sure if we only push a single change or not we may as well use a more generic message.